lib/matplotlib/projections/geo.py

```diff
@@ -190,7 +190,7 @@ def set_longitude_grid(self, degrees):
         """
         Set the number of degrees between each longitude grid.
         """
-        number = (360.0 / degrees) + 1
+        number = int(360.0 / degrees) + 1
```

Comment:
I think the confusion was whether it's working correctly. I'm not sure what rounding behavior we want, but the `+ 1` is needed to work with the call to linspace below, since we're removing the first and last points. Would probably make more sense as:

```python
number = np.round(360 / degrees)
ticks = np.linspace(-nppi, np.pi, number, endpoint=False)[1:]
self.axis.set_major_locator(FixedLocator(ticks))
```